sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pandas as pd
+import time
+from typing import types
+from tqdm.auto import tqdm
+from sktime.utils.validation.forecasting import check_y
+from sktime.utils.validation.forecasting import check_cv
+from sktime.forecasting.base import ForecastingHorizon
+from sktime.performance_metrics.forecasting import smape_loss
+
+__author__ = ["Martin Walter"]
+__all__ = ["evaluate"]
+
+
+def evaluate(
+    forecaster, cv, y, X=None, strategy="refit", scoring=smape_loss, return_data=False
+):
+    """Evaluate forecaster using cross-validation
+
+    Parameters
+    ----------
+    forecaster : sktime.forecaster
+        Any forecaster
+    cv : sktime.SlidingWindowSplitter or sktime.ExpandingWindowSplitter
+        Splitter of how to split the data into test data and train data
+    y : pd.Series
+        Target time series to which to fit the forecaster.
+    X : pd.DataFrame, optional (default=None)
+        Exogenous variables
+    strategy : str, optional
+        Must be "refit" or "update", by default "refit". The strategy defines
+        whether forecaster is only fitted on the first train window data and
+        then updated or always refitted.
+    scoring : function, optional
+        A score function that takes y_pred and y_test as arguments,
+        by default smape_loss
+    return_data : bool, optional
+        Returns three additional columns in the DataFrame, by default False.
+        The cells of the columns contain each a pd.Series for y_train,
+        y_pred, y_test.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame that contains several columns with information regarding each
+        refit/update and prediction of the forecaster.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.performance_metrics.forecasting import evaluate
+    >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> y = load_airline()
+    >>> forecaster = NaiveForecaster(strategy="drift", sp=12)
+    >>> cv = ExpandingWindowSplitter(
+        initial_window=24,
+        step_length=12,
+        fh=[1,2,3,4,5,6,7,8,9,10,11,12]
+        )
+    >>> evaluate(forecaster=forecaster, y=y, cv=cv)
+    """
+    cv = check_cv(cv)
+    y = check_y(y)
+    _check_strategies(strategy)
+    assert isinstance(evaluate, types.FunctionType), "scoring must be a function"
```

Comment:
Can we make use of the `check_scoring` function [here](https://github.com/alan-turing-institute/sktime/blob/45d88931288f076f7e42043399567b77130406b2/sktime/utils/validation/forecasting.py#L309)? This would also consistently handle the default argument via `scoring=None`